### DISCUSSION TITLE
FieldArray validation overrides nested fields validation

### DISCUSSION THREAD

**Discussion by pavelspichonak** [react-hook-form/react-hook-form#8990] [Q&A] [answered]
I'm using `useFieldArray` to render array of objects. Array must have its own validation(min/max) and properties of array objects must have required and other validations. Unfortunately array validations override array object validations. In other words until array validations pass you do not see validation errors of array objects.

What I would like to have is to see array validations errors and array objects validations errors at the same time but I have no idea how to achieve it.

[Here is example](https://stackblitz.com/edit/react-hook-form-7-dynamic-form-example-ebwbtr?file=App.jsx)

Press submit button(Buy Tickets)

Email and Name fields are required. But you will not see validation error for these fields until you add at least two tickets using Add ticket button. Tickets array has **yup** validation `.min(2, 'min 2!')`

**Comment by pavelspichonak** [comment]
Hi @bluebill1049. Is it a bug or expected behaviour? Maybe you know how to handle this case?

**Comment by Moshyfawn** [comment, marked as answer]
I believe, it's related to [v7.34.0](https://github.com/react-hook-form/react-hook-form/releases/tag/v7.34.0). It adds the new field array `rules` validation API, but [the resolvers package](https://react-hook-form.com/api/usefieldarray/api/useform/#validationResolver) is yet to support the feature.
RHF [useFieldArray docs ref](https://react-hook-form.com/api/usefieldarray): 
> [Resolvers](https://react-hook-form.com/api/usefieldarray/api/useform/#validationResolver) are yet to support useFieldArray root level validation.

**Reply by pavelspichonak** [reply]
Thanks. And looks like we can't use them both(yup validation for array objects and rules field for arrays)... @Moshyfawn maybe you know are there plans to add this feature in yup resolver?

**Reply by Moshyfawn** [reply]
Validation schema can't be used with field level validation rules (`<Controller rules={{}} />` / `register("fieldName", { required: true })`). If a resolver is been provided, the field level rules are ignored.

We're planning on catching the resolver API up to speed with the updated field array error shape / API, but we still need to find time to do it

**Reply by pavelspichonak** [reply]
Got it. Thanks a lot!
